charts/vector/templates/_pod.tpl

```diff
@@ -15,10 +22,14 @@ containers:
 {{- end }}
     image: "{{ .Values.image.repository }}:{{ .Values.image.tag | default .Chart.AppVersion }}"
     imagePullPolicy: {{ .Values.image.pullPolicy }}
+{{- with .Values.args }}
     args:
-      - --config-dir
-      - "/etc/vector/"
+    {{- toYaml . | nindent 6 }}
```

Comment:
Allows for users to add more options/flags or replace our default config dirs (pipelines soon). This should be less brittle as a user needs to replace each item in the list, so us changing defaults wouldn't impact someone already adjusting the args (vs appending additional args to the end of the list)